tests/test_estimators/test_demographic_parity.py

```diff
@@ -0,0 +1,113 @@
+import warnings
+
+import pytest
+import numpy as np
+from cvxpy import SolverError
+from sklearn.linear_model import LogisticRegression
+
+from sklego.common import flatten
+from sklego.linear_model import DemographicParityClassifier, FairClassifier
+from sklego.metrics import p_percent_score
+from tests.conftest import general_checks, nonmeta_checks, classifier_checks
+
+
+@pytest.mark.parametrize(
+    "test_fn", flatten([general_checks, nonmeta_checks, classifier_checks])
+)
+def test_standard_checks(test_fn):
+    trf = DemographicParityClassifier(
+        covariance_threshold=None,
+        C=1,
+        penalty="none",
+        sensitive_cols=[0],
+        train_sensitive_cols=True,
+    )
+    test_fn(DemographicParityClassifier.__name__, trf)
+
+
+def _test_same(dataset):
+    X, y = dataset
+    if X.shape[1] == 1:
+        # If we only have one column (which is also the sensitive one) we can't fit
+        return True
+
+    sensitive_cols = [0]
+    X_without_sens = np.delete(X, sensitive_cols, axis=1)
+    lr = LogisticRegression(penalty="none", solver="lbfgs")
+    fair = DemographicParityClassifier(
+        covariance_threshold=None, sensitive_cols=sensitive_cols, penalty="none"
+    )
+    try:
+        fair.fit(X, y)
+    except SolverError:
```

Comment:
is there a good reason for this? something tells me that there should be no `SolverError` for the `random_xy_dataset_clf` datasets

Reply:
Absolutely right! I removed it